### PR TITLE
Add 600 subjects

### PR DESCRIPTION
**Pull request by alicefuzier**
For #2523

### CODE REVIEW

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraConceptSubjects.scala [hunk 1]
```diff
@@ -51,7 +48,15 @@ trait SierraSubjects extends MarcUtils with SierraConcepts {
   //      Note that only concepts from subfield $a are identified; everything
   //      else is unidentified.
   //
-  private def getSubjectsForMarcTag(bibData: SierraBibData, marcTag: String) = {
+  def getSubjectswithAbstractConcepts(
+    bibData: SierraBibData): List[Subject[MaybeDisplayable[AbstractConcept]]] =
+    getSubjectsForMarcTag(bibData, "650") ++ getSubjectsForMarcTag(
+      bibData,
+      "648") ++ getSubjectsForMarcTag(bibData, "651")
```
Comment (alexwlchan) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraConceptSubjects.scala]: nit: the auto formatting has mangled this, making it look like the old `getSubjects` would make it easier to read.

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraPersonSubjects.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  //
+  // Use MARC field "600" where the second indicator is not 7.
+  //
+  // Within this MARC tag we have one concept which we always type as a Person:
+  //  - subfield $a populates the person label
+  //  - subfield $b populates the person numeration
+  //  - subfield $c populates the person prefixes
+  //
+  // The label is constructed concatenating subfields $a, $b, $c, $d, $e,
+  // where $d and $e represent the persons dates and roles respectively
```
Comment (alexwlchan) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraPersonSubjects.scala]: nit: should be an apostrophe in "person's".

File: ontologies/WIP/sierratransformable [hunk 3]
```diff
@@ -327,7 +327,7 @@
    /* Populate wwork:subjects people and organisations. Rules:
    * 
    * 1. For all bibliographic records use "marcTag": "600", "610". 
-   * 2. Populate platform "label" for "600" and "610" concatenated with subfields "a", "c", "d" "e"  type as "Subject". 
+   * 2. Populate platform "label" for "600" and "610" concatenated with subfields "c", "a", "b", "d", "e"  type as "Subject".
```
Comment (alexwlchan) [inline ontologies/WIP/sierratransformable]: Is the corresponding transformer change not done yet, or have I just missed it?

File: sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Agent.scala [hunk 4]
```diff
@@ -1,6 +1,18 @@
 package uk.ac.wellcome.models.work.internal
 
-sealed trait AbstractAgent {
+sealed trait AbstractConcept {
+  val label: String
+}
+
+sealed trait Bah extends AbstractConcept
```
Comment (alexwlchan) [inline sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Agent.scala]: \*raises eyebrow*

Comment (alicefuzier) [reply sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Agent.scala]: Yeah, I know, it was a WIP

File: ontologies/WIP/sierratransformable (hunk 3 above)
Comment (alicefuzier) [reply ontologies/WIP/sierratransformable]: The transformer change for 610 will come in another P, this is only for 600. ALso the rules for 610 are probably going to be different from what the sierratransformable file says (check #2556 ). But we need input from @silveroliver about those

**Comment by kenoir** [comment]
Looks good to me, but @alexwlchan might be better off approving.

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraPersonSubjects.scala [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+trait SierraPersonSubjects extends MarcUtils with SierraAgents {
+
+  // Populate wwork:subject
+  //
+  // Use MARC field "600" where the second indicator is not 7.
+  //
+  // Within this MARC tag we have one concept which we always type as a Person:
+  //  - subfield $a populates the person label
+  //  - subfield $b populates the person numeration
+  //  - subfield $c populates the person prefixes
```
[the remark this answers is not in the document]
Comment (alicefuzier) [reply catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/subjects/SierraPersonSubjects.scala]: yeah good point

**Pull request merged**